src/pages/invoices/common/hooks/useProductColumns.ts

```diff
@@ -16,47 +16,133 @@ export function useProductColumns() {
   const company = useCurrentCompany();
   const [columns, setColumns] = useState<string[]>([]);
 
+  const getColumnIndex = (
+    columnKey: string,
+    pdfColumns: string[],
+    currentColumns: string[]
+  ) => {
+    const columnIndex = pdfColumns.findIndex(
+      (variable) => variable === columnKey
+    );
+
+    const adjustedColumnIndex =
+      columnIndex > -1 ? columnIndex : currentColumns.length;
+
+    return adjustedColumnIndex;
+  };
+
   useEffect(() => {
-    // We need to clone the product columns to local object,
-    // because by default it's frozen.
-    let variables: string[] =
+    let updatedVariables: string[] =
       clone(company?.settings.pdf_variables.product_columns) || [];
 
-    // Local object is needed because we want to spread tax columns in case they're enabled.
-    if (variables.includes('$product.tax')) {
-      const taxes: string[] = [];
-      const enabledTaxRates = company?.enabled_item_tax_rates || 0;
+    let pdfVariables =
+      clone(company?.settings.pdf_variables.product_columns) || [];
 
-      if (enabledTaxRates > 0) {
-        taxes.push('$product.tax_rate1');
-      }
+    const numberOfPdfVariables = pdfVariables.length;
 
-      if (enabledTaxRates > 1) {
-        taxes.push('$product.tax_rate2');
-      }
+    if (!updatedVariables.includes('$product.item')) {
+      updatedVariables.splice(0, 0, '$product.item');
+      pdfVariables.splice(0, 0, '$product.item');
```

Comment:
Please don't use a splice, rather split, filter, or spread the existing.

Reply:
I have created a separate function for inserting variables that does not use the `splice` function to make the code clearer and easier to read.